internal/meshdetect/detector.go

```diff
@@ -213,25 +221,94 @@ func isPodInitContainerInjected(meshKind MeshKind, pod *corev1.Pod) bool {
 
 // DetectServiceDistribution detects how many services are running under each mesh.
 func (d *Detector) DetectServiceDistribution(ctx context.Context) (*ServiceDistributionResults, error) {
+
 	ret := &ServiceDistributionResults{
 		MeshDistribution: map[MeshKind]int{},
 	}
 
 	// list all services
-	serviceList := &corev1.ServiceList{}
-	err := d.Client.List(ctx, serviceList)
-	if err != nil {
-		d.Logger.Error(err, "failed to list all services in cluster")
-		return nil, err
+	serviceList := []*corev1.Service{}
```

Comment:
About this whole block: how about we put those listing snippets in separate `func`s with parameters (like pageSize)? It would make `DetectServiceDistribution()` way more readable